### ISSUE TITLE
Weird interaction with `LiteralString` and f-strings

### ISSUE THREAD

**Issue by otonnesen** [microsoft/pyright#4106]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
Conditional variable binding interacts unexpectedly with LiteralString f-strings.

**To Reproduce**
See attached code snippet.

**Expected behavior**
I expected f-string interpolation and string concatenation to be treated similarly.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing_extensions import LiteralString


baz: LiteralString

if 1 == 1:
    foo = "foo"
    baz = f"I am {foo}"  # OK
else:
    foo = "bar"
    baz = f"I am {foo}"  # OK

baz = f"I am {foo}"  # Expression of type "str" cannot be assigned to declared type "LiteralString"  
baz = "I am " + foo  # OK

```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Running CLI pyright v1.1.277

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.278, which I just published. It will also be included in a future release of pylance.
